### DISCUSSION TITLE
How to use 'Transaction Log'?

### DISCUSSION THREAD

**Discussion by LiJie20190102** [JanusGraph/janusgraph#4333] [Q&A]
I created a test using ‘Transaction Log’, but it did not meet my expectations. Is there a problem with my usage?
232 lines have no execution effect.
![image](https://github.com/JanusGraph/janusgraph/assets/53458004/1bd1c733-810a-4eed-99dc-46e4f6e95f25)

**Comment by li-boxuan** [comment]
Log processor runs in a separate thread. Try adding a `Thread.sleep` before the test exits.

**Reply by LiJie20190102** [reply]
![image](https://github.com/JanusGraph/janusgraph/assets/53458004/c26118da-ca3f-4e6b-a09e-e5d8cf1e7e51)
Is this how it is used? It still has no effect

**Reply by li-boxuan** [reply]
Try increasing to 30 seconds?

**Reply by LiJie20190102** [reply]
60 seconds still has no effect ,Please help me check if there are any issues with the code

**Reply by li-boxuan** [reply]
Please post your full code here

**Reply by LiJie20190102** [reply]
@Test
  public void testLogTransaction() throws InterruptedException {

      StandardJanusGraph g = (StandardJanusGraph)ConfiguredGraphFactory.open("graph_test2");


      JanusGraphTransaction start =
          g.buildTransaction().checkExternalVertexExistence(false).logIdentifier("addedPerson").start();

      // vertices
       start.addVertex(T.id, "lisi2", T.label, "people","name","aaaaa");
      start.commit();
      System.out.println("=============ok");

      LogProcessorFramework logProcessor = JanusGraphFactory.openTransactionLog(g);

      logProcessor.addLogProcessor("addedPerson").
          setProcessorIdentifier("addedPersonCounter").
          setStartTime(Instant.MIN).setRetryAttempts(1).
          addProcessor(new ChangeProcessor() {
              @Override
              public void process(JanusGraphTransaction tx, TransactionId txId, ChangeState changeState) {
                  Set<JanusGraphVertex> vertices = changeState.getVertices(Change.ADDED);
                  for (JanusGraphVertex vertex : vertices) {
                      if (vertex.label().equals("people")) {
                          System.out.println("vertex==1=="+vertex);
                      }
                  }
              }
          }).
          build();

      Thread.sleep(60000);

  }

**Reply by li-boxuan** [reply]
weird, I haven't used this feature for a long time but I have used it in production before...

**Comment by LiJie20190102** [comment]
@li-boxuan 
Meanwhile, in (https://docs.janusgraph.org/advanced-topics/janusgraph-bus/), I noticed that there seems to be a different explanation for 'Transaction Log' and instead recommended me to use 'Trigger Log'. I'm a bit confused. Is there a more detailed introduction?

**Reply by li-boxuan** [reply]
This is a known issue: https://github.com/JanusGraph/janusgraph/issues/3128

You could pretty much ignore what janusgraph-bus said.

**Reply by LiJie20190102** [reply]
Okay, based on the code (https://github.com/JanusGraph/janusgraph/discussions/4333#discussioncomment-8824264) I provided, have you found the reason why it cannot take effect

**Reply by li-boxuan** [reply]
Nope... I suggest you ask in the mailing list

**Reply by li-boxuan** [reply]
There's also a test in JanusGraphTest.java. You might want to use debugger to test that.

**Reply by LiJie20190102** [reply]
Okay, an email has been sent for assistance. I referred to org.janusgraph.graphdb.JanusGraphTest#simpleLogTestWithFailure, but I didn't get the result I wanted either
